packages/catalog-process/src/services/catalogService.ts

```diff
@@ -617,16 +617,12 @@ async function innerAddDocumentToEserviceEvent(
   descriptorId: DescriptorId,
   documentSeed: catalogApi.CreateEServiceDescriptorDocumentSeed,
   templateVersionId: EServiceTemplateVersionId | undefined,
-  readModelService: ReadModelService,
-  { authData, correlationId }: WithLogger<AppContext>
-): Promise<{ eService: EService; event: CreateEvent<EServiceEvent> }> {
-  await assertRequesterIsDelegateProducerOrProducer(
-    eService.data.producerId,
-    eService.data.id,
-    authData,
-    readModelService
-  );
```

Comment:
No need to this check to be here inside an inner function. 
Plus, it was called multiple time inside routes that needs to upload more than one file.